### ISSUE TITLE
Runtime error when using --build-optimizer, happens in --prod mode only

### ISSUE THREAD

**Issue by mrjmd** [angular/angular-cli#7190]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.
![screen shot 2017-07-28 at 9 21 22 am](https://user-images.githubusercontent.com/2111857/28726478-3e815176-7376-11e7-9c60-53ffaaee2fe0.png)

### Repro steps.
`git clone git@github.com:contentacms/contenta_angular.git`
`npm install`
`ng serve --prod --bo`

### The log given by the failure.
![screen shot 2017-07-28 at 9 24 00 am](https://user-images.githubusercontent.com/2111857/28726575-99ea4522-7376-11e7-8d79-a930403dcb62.png)

### Mention any other details that might be useful.
The build seems fine using `ng serve --prod --bo`, but this error occurs at runtime. The build and runtime work fine if we use `ng serve --prod` alone, or if we use `ng serve --aot --bo`.

**Comment by filipesilva** [comment]
Thank you for the report and repro, will investigate.

**Comment by xban1x** [comment]
I do not recieve exactly the same error, but i noticed that i can use AOT with BO or AOT with PROD but you cannot mix PROD and BO.

**Comment by filipesilva** [comment]
I looked into it a bit today. Wasn't able to reach a definite conclusion, but this is what I found:
- only happens with both `--prod --build-optimizer`, `--aot --build-optimizer` doesn't trigger it. So it seems to be related to Uglify as well.
- seems to be related to an `__extends` related optimization
- seems related to angular2-jsonapi and contenta_angular_service

I have to investigate further.

**Comment by filipesilva** [comment]
Just tested and this is fixed on `1.3.0-rc.5`.

**Comment by mrjmd** [comment]
This is indeed fixed, thanks! Unfortunately 1.3.0-rc.5 seems to have broken our service worker (worker-basic is still there, but sw-register bundle does not reference it) but I will open a new issue about that.

**Comment by mrjmd** [comment]
Just filed #7280 regarding the service worker issue.
